MatrixSDKTests/MXAuthenticationSessionTests.swift

```diff
@@ -0,0 +1,122 @@
+// 
+// Copyright 2020 The Matrix.org Foundation C.I.C
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import XCTest
+
+import MatrixSDK
+
+class MXAuthenticationSessionTests: XCTestCase {
+
+    override func setUpWithError() throws {
+        // Put setup code here. This method is called before the invocation of each test method in the class.
+    }
+
+    override func tearDownWithError() throws {
+        // Put teardown code here. This method is called after the invocation of each test method in the class.
+    }
+
+    func testParsing() throws {
+        
+        let json: [String: Any] = [
+            "completed": [],
+            "session": "2134",
+            "flows": [
+                ["type": "m.login.password",
+                 "stages": [],
+                ],
+                ["type": "m.login.sso",
+                 "stages": [],
+                 "identity_providers": [
+                    ["id": "gitlab",
+                     "name": "GitLab"
+                    ],
+                    ["id": "github",
+                     "name": "GitHub",
+                     "icon": "https://github/icon.png"
+                    ]
+                 ]
+                ],
+                ["type": "m.login.cas",
+                 "stages": [],
+                 "identity_providers": []
+                ]
+            ],
+            "params": [
+            ]
+        ]
+        
+        let authenticationSession = MXAuthenticationSession(fromJSON: json)
+                       
+        guard let flows = authenticationSession?.flows else {
+            XCTFail("flows shouldn not be nil")
+            return
+        }
+        
+        XCTAssertEqual(flows.count, 3)
+        
+        if let ssoFlow = flows.first(where: { $0.type == kMXLoginFlowTypeSSO }) {
+            
+            if let loginSSOFlow = ssoFlow as? MXLoginSSOFlow {
```

Comment:
This is what i wanted to avoid on above comment.

Reply:
He have to differentiate `m.login.sso` from `m.login.cas` they are both the same kind of flow but not exactly the same :)

Reply:
This will continue as the same, explained in https://github.com/matrix-org/matrix-ios-sdk/pull/965#discussion_r536027445